appinventor/components/src/com/google/appinventor/components/runtime/Ev3GyroSensor.java

```diff
@@ -103,15 +106,23 @@ public double GetSensorValue() {
    * Specifies the mode of the sensor.
    */
   @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_LEGO_EV3_GYRO_SENSOR_MODE,
-                    defaultValue = DEFAULT_SENSOR_MODE_STRING)
+                    defaultValue = "angle")
   @SimpleProperty
-  public void Mode(String modeName) {
-    String functionName = "Mode";
-    try {
-      setMode(modeName);
-    } catch(IllegalArgumentException e) {
-      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_ARGUMENT, functionName);
+  public void Mode(@Options(GyroSensorMode.class) String modeName) {
+    // Make sure modeName is a valid GyroSensorMode.
+    GyroSensorMode gyroMode = GyroSensorMode.fromUnderlyingValue(modeName);
+    if (gyroMode == null) {
+      form.dispatchErrorOccurredEvent(
```

Comment:
Need to either return after dispatching the error or put the `setMode` call in an `else` statement.